src/database/utilities/createRecord.js

```diff
@@ -206,22 +206,25 @@ const createRequisitionItem = (database, requisition, item, dailyUsage) => {
  * Create a new stocktake.
  *
  * @param   {Realm}      database
- * @param   {User}       user      User creating stocktake.
+ * @param   {User}       user             User creating the stocktake.
+ * @param   {string}     stocktakeName    What to name the stocktake
+ * @param   {Program}    program          Program record to associate with the stocktake
  * @return  {Stocktake}
  */
-const createStocktake = (database, { createdBy, program, name }) => {
+const createStocktake = (database, user, stocktakeName, program) => {
   const date = new Date();
-  const { stocktakeText } = generalStrings;
-  const defaultName = `${stocktakeText} - ${name} - ${formatDateAndTime(date, 'slashes')}`;
+  const serialNumber = getNextNumber(database, STOCKTAKE_SERIAL_NUMBER);
+  const title = program ? program.name : generalStrings.stocktake;
+  const defaultName = `${title} - ${formatDateAndTime(date, 'slashes')}`;
```

Comment:
I guess this is the core code change and the rest are just code refactors. 
Looks good!

Reply:
To some degree. The old both `stocktakeText` and `name` were coming through undefined so the refactoring was in part to resolve that.

Also I don't like passing of objects as args and destructuring the props you want out of them. It invites just passing through everything rather than just what you need (i.e. we'd see the entire state of `byProgramModal.js` showing up here before).